lib/sensor.js

```diff
@@ -223,6 +231,20 @@ function Sensor(opts) {
         return this.constrained;
       }
     },
+    freq: {
+      get: function() {
+        return priv.get(this).freq;
+      },
+      set: function(newFreq) {
+        priv.set(this, {
+          freq: newFreq
+        });
```

Comment:
I have not worked with javascript Map previously.  Transposing from other languages, I was thinking that 'this' was a context, not a key.  Instead, priv is a file scope variable (class static in other languages), and the instance ('this') is the key.  I actually implemented a similar structure in older (es5) javascript using a serial number incremented by the constructor as a key to create class and instance private storage.  Map and using a reference to the instance object makes it cleaner.  Time to write a test function, to make sure I get the capabilities straight.